storagemanager/storage/views.py

```diff
@@ -23,49 +23,65 @@ class ShowAllListView(ListView):
-    ordering = ['category', 'type']
+    ordering = ['category', 'type', 'model']
 
 def show_items_from_category(request):
     categories_set = set()                      
     for item in Item.objects.all():
         categories_set.add(item.category)
-    sorted_items = sorted(Item.objects.filter(category=request.POST.get('category')), key=attrgetter('type', 'model'))
+    sorted_items = Item.objects.filter(category=request.POST.get('category')).order_by('type', 'model')
     context = {
         'items_to_show': sorted_items, 
         'categories': categories_set
         }
     return render(request, 'storage/show_items_from_category.html', context)
 
 def delete_item(request):
-    context = {'all_items': sorted(Item.objects.all(), key=attrgetter('category', 'type', 'model'))}
+    context = {'all_items': Item.objects.all().order_by('category', 'type', 'model')}
     return render(request, 'storage/delete_item.html', context)
 
 def delete_item_confirm(request):
-    try:
-        context = {
-            'item_to_delete': str(Item.objects.filter(id=request.POST.get('id')).first()), 
-            'id': request.POST.get('id')
-            }
-        if context.get('item_to_delete').startswith('None'):
-            messages.warning(request, f'Item with that ID could not be found.')
-            return redirect('delete-item')
-        else:
-            return render(request, 'storage/delete_item_confirm.html', context)
-    except ValueError:
+    if request.POST.get('id') == '':
         messages.warning(request, f'Please type a number')
         return redirect('delete-item')
+    elif 'e' in request.POST.get('id'):
+        messages.warning(request, f'Exponent numbers are not allowed')
+        return redirect('delete-item')
+    else:
+        item_id = request.POST.get('id')
+
+    if item_id == 'Choose_item':
+        messages.warning(request, f'Please choose an item from the list.')
+        return redirect('delete-item')
+
+    try:
+        item_to_delete = Item.objects.get(id=item_id)
+    except ObjectDoesNotExist:
+        item_to_delete = None
+    context = {
+        'item_to_delete': item_to_delete, 
+        'id': item_id
+        }
+    if context.get('item_to_delete') == None:
+        messages.warning(request, f'Item with that ID could not be found.')
+        return redirect('delete-item')
+    else:
+        return render(request, 'storage/delete_item_confirm.html', context)
+
 
 def delete_item_after_confirm(request):
-    item_var = Item.objects.filter(id=request.POST.get('id')).first()
-    Item.objects.filter(id=request.POST.get('id')).first().delete()
-    messages.success(request, f'Item: {item_var.type} {item_var.model}, SN: {item_var.serial_number} has been deleted')
+    item_to_delete = Item.objects.get(id=request.POST.get('id'))
+    item_to_delete_type = item_to_delete.type
+    item_to_delete_model = item_to_delete.model
+    item_to_delete_serial_number = item_to_delete.serial_number
```

Comment:
There is not much value in these assignments (we basically exchange one `.` for `_`). Maybe just use `type`, `model`, `serial_number`? There are no multiple items/types/models/numbers, no chances we will mix it with something different here, right?